### DISCUSSION TITLE
Another `linkname` usage

### DISCUSSION THREAD

**Discussion by HeRaNO** [opencontainers/runc#4293] [General]
https://github.com/golang/go/blob/master/src/internal/poll/fd_poll_runtime.go#L165-L179 shows we use `IsPollDescriptor` as a `linkname`, and we used here https://github.com/opencontainers/runc/blob/main/libcontainer/utils/utils_unix.go#L112-L119.

So maybe we need a `-ldflags=-checklinkname=0` to make the compiling stage happy on Go 1.23. Or we can make a change to avoid using `linkname`.

UPD: Fortunately it's added at runc 1.2.0-rc1. We can eliminate it to avoid being on the hall of shame.

**Comment by HeRaNO** [comment]
Not quite sure why the compiling stage is ok on `tip`... Maybe I made a mistake (is now we have a so-called `Handshake` situation?). Comments welcome.

**Comment by cyphar** [comment]
We are aware of this and [have an upstream proposal](https://github.com/golang/go/issues/67639) to avoid the need for `go:linkname` for getting whether file descriptors are internal to the Go runtime.

(For what it's worth, I don't really care about their "hall of shame". All of our `go:linkname` usages are to workaround stdlib bugs or limitations of Go that make Go unusable as a "systems language" without hacks.)
